trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoSplitSource.java

```diff
@@ -25,10 +25,14 @@ public GravitinoSplitSource(ConnectorSplitSource connectorSplitSource) {
 
   @Override
   public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize) {
-    ConnectorSplitBatch batch = connectorSplitSource.getNextBatch(maxSize).join();
-    List<ConnectorSplit> list =
-        batch.getSplits().stream().map(GravitinoSplit::new).collect(Collectors.toList());
-    return CompletableFuture.completedFuture(new ConnectorSplitBatch(list, batch.isNoMoreSplits()));
```

Comment:
`connectorSplitSource.getNextBatch(maxSize).join()` is synchronized. It block processing in the schedule thread.

Reply:
why block 1 min? means split will take nearly 1 min?

Reply:
Not 1 min. about 20s , it have 20 splits in that query.